06_boolean_expressions.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# Implementation for RubyMonk Primer Problem 6 Hiring Programmers - Boolean Expressions in Ruby
+# Problem Statement:
+# Let us say you are trying to recruit team-members for your new startup! Given a candidate,
+#  you need an expression that will tell you whether they fit into certain types. This is
+#  how a candidate object would look:
+
+# candidate.years_of_experience = 4
+#   candidate.github_points = 293
+#   candidate.languages_worked_with = ['C', 'Ruby', 'Python', 'Clojure']
+#   candidate.applied_recently? = false
+#   candidate.age = 26
+
+# We are looking to hire experienced Ruby programmers. Our ideal candidate has 2 or more years
+# of experience, but some programmers become really good even before that. We'll consider their
+# Github points (a nice indicator of a good programmer), and even if they are not experienced,
+# candidates with 500 Github points or more can apply. And there is one more catch: Ruby being
+# a cool and awesome language, a lot of smart youngsters are very good at it. We love those kids,
+# but for this particular job we'd rather have them study at school than work. Let us filter out
+# candidates who are younger than 15. Also we don't want to consider candidates who applied
+# recently for this opening.
+
+# Implementing the problem's candidate class
+class Candidate
+  attr_accessor :years_of_experience, :github_points, :languages_worked_with, :applied_recently, :age
+
+  def initialize(
+    years_of_experience: 0,
+    github_points: 0,
+    languages_worked_with: [],
+    applied_recently: false,
+    age: 0
+  )
+
+    @years_of_experience = years_of_experience
+    @github_points = github_points
+    @languages_worked_with = languages_worked_with
+    @applied_recently = applied_recently
+    @age = age
+  end
+
+  def applied_recently?
+    @applied_recently
+  end
+end
+
+# Global Methods
+def excluded_candidate?(candidate)
+  (candidate.age < 15) || candidate.applied_recently?
+end
+
+def experienced_programmer?(candidate)
+  (candidate.years_of_experience >= 2) || (candidate.github_points >= 500)
+end
+
+def ruby_experience?(candidate)
+  candidate.languages_worked_with.include? 'Ruby'
+end
+
+def valid_ruby_candidate?(candidate)
+  return false if excluded_candidate?(candidate)
+
+  experienced_programmer?(candidate) && ruby_experience?(candidate)
+end
+
+# FYI: RubyMonk's solution
+def their_solution(candidate)
```

Comment:
I think your solution is better because it separates everything out into methods

Reply:
Agree! 💯